src/server/pfs/cmds/cmds.go

```diff
@@ -310,6 +310,39 @@ $ pachctl list-commit foo master --from XXX
 	listCommit.Flags().IntVarP(&number, "number", "n", 0, "list only this many commits; if set to zero, list all commits")
 	rawFlag(listCommit)
 
+	printCommitIter := func(commitIter client.CommitInfoIterator) error {
+		if raw {
+			for {
+				commitInfo, err := commitIter.Next()
+				if err == io.EOF {
+					return nil
+				}
+				if err != nil {
+					return err
+				}
+				if err := marshaller.Marshal(os.Stdout, commitInfo); err != nil {
+					return err
+				}
+			}
+		}
+		writer := tabwriter.NewWriter(os.Stdout, 20, 1, 3, ' ', 0)
+		for {
+			commitInfo, err := commitIter.Next()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return err
+			}
+			pretty.PrintCommitInfoHeader(writer)
+			pretty.PrintCommitInfo(writer, commitInfo)
+			if err := writer.Flush(); err != nil {
+				return err
+			}
```

Comment:
I think you need to call `writer.Init` in here too otherwise it will use the alignment of the first call to `Flush` for every subsequent print.